sktime/forecasting/model_evaluation/_functions.py

```diff
@@ -137,40 +200,45 @@ def _evaluate_window(
             forecaster.update(y_train, X_train, update_params=update_params)
         fit_time = time.perf_counter() - start_fit
 
-        pred_type = {
+        # predict based on metrics
+        pred_type = {  # tuple indicates (method for predict, naming)
             "pred_quantiles": "predict_quantiles",
             "pred_interval": "predict_interval",
             "pred_proba": "predict_proba",
-            None: "predict",
+            "pred": "predict",
         }
-        # predict
-        start_pred = time.perf_counter()
-
-        if hasattr(scoring, "metric_args"):
-            metric_args = scoring.metric_args
-        else:
-            metric_args = {}
+        for scitype in scoring:
+            for metric in scoring.get(scitype):
+                if hasattr(metric, "metric_args"):
+                    metric_args = metric.metric_args
+                else:
+                    metric_args = {}
+                methodname = pred_type[scitype]
+                method = getattr(forecaster, methodname)
 
-        if hasattr(scoring, "get_tag"):
-            scitype = scoring.get_tag("scitype:y_pred", raise_error=False)
-        else:
-            # If no scitype exists then metric is not proba and no args needed
-            scitype = None
+                start_pred = time.perf_counter()
+                y_pred = method(fh, X_test, **metric_args)
```

Comment:
I did not see the `metrics_args` tag in the registry tag, so I am unsure when it is used or if it will ever be used. I suppose this is another defensive mechanism you implemented for metrics that require extra arguments to be passed to the `.predict` method. Could you please correct me if I am wrong? Also, could you inform me of when these cases may occur?

Reply:
ignore me, I found such case in PinballLoss metrics.

Reply:
I think the way in which I would solve this by using the `metric_args` of the first metric of each scitype and pass it to the `.predict`  then we cache the prediction so other metrics of the same type can just use it for computing their values.

Reply:
I'm confused about the `metric_args` too, it seems wrong. I'm not sure who wrote this, it might be it was me but it still doesn't seem right at the moment to me.

You said you understood an example of this - can you kindly expand and spell out the example?

Reply:
ok so from the way I understand it, for `predict_proba`, `predict_interval`, and `predict_quantiles`, they have additional arguments compared to point forecast, e.g predict_proba has `marginal`, for quantiles `alpha` while for predict_interval has `coverage`. the current interface uses `metric_args` attribute from a metric to control those arguments but the thing is, from what I know,  only **PinballLoss** has `metrics_args` specifying alpha hence how niche the case is.

Reply:
I see - to me, it sounds a bit like duplicative design, i.e., arguments which should be passed at `predict_method` and not at the metric have somehow gotten attached to the metric and then need to be passed around by external logic, which is bad design-wise (high coupling).

I think this is a historical accident from the time we started working on the proba metrics, I guess one could safely ignore it, but cleaning up that particular issue seems out of scope for this PR.

So, as out of scope, I suppose we should leave it in for now, but might want to return later to reduce code entropy in the interfaces here.

Reply:
tracked here #5279